test/unit/search/search_test.cpp

```diff
@@ -106,6 +106,20 @@ TYPED_TEST(search_test, multiple_queries)
     EXPECT_RANGE_EQ(search(queries, this->index, cfg) | position, (std::vector{0, 0, 4}));
 }
 
+TYPED_TEST(search_test, parallel_queries)
+{
+    std::vector<std::vector<seqan3::dna4>> const queries{100, {"ACGTACGTACGT"_dna4}};
+
+    seqan3::configuration const cfg = seqan3::search_cfg::max_error_total{seqan3::search_cfg::error_rate{.0}} |
+                                      seqan3::search_cfg::max_error_substitution{seqan3::search_cfg::error_rate{.0}} |
+                                      seqan3::search_cfg::max_error_insertion{seqan3::search_cfg::error_rate{.0}} |
+                                      seqan3::search_cfg::max_error_deletion{seqan3::search_cfg::error_rate{.0}} |
+                                      seqan3::search_cfg::parallel{2};
+
+    EXPECT_RANGE_EQ(search(queries, this->index, cfg) | query_id, std::views::iota(0, 100));
+    EXPECT_RANGE_EQ(search(queries, this->index, cfg) | position, std::vector(100, 0));
```

Comment:
Just as a note:
`std::vector(100, 0)` cannot be `std::vector{100, 0}`, because then it would use the initialiser list constructor (=[100,0]) instead of the "count, value" constructor (=[0,0,...,0]).